token/core/common/ppm.go

```diff
@@ -0,0 +1,143 @@
+	// PublicParams returns the public parameters
+	PublicParams() ([]byte, error)
+}
+
+type PublicParamsManager[T driver.PublicParameters] struct {
+	Logger *flogging.FabricLogger
+
+	PublicParamsDeserializer PublicParamsDeserializer[T]
+	publicParameters         T
+	// the vault
+	Vault Vault
+	// label of the public params
+	PPLabel string
+	// Mutex is used to control access to the public parameters
+	Mutex sync.RWMutex
+	// Callbacks are a set of functions to be called when the parameters are set
+	Callbacks []SetPublicParametersCallbackFunc
+}
+
+func NewPublicParamsManager[T driver.PublicParameters](
+	logger *flogging.FabricLogger,
+	PublicParamsDeserializer PublicParamsDeserializer[T],
+	PPLabel string,
+	vault Vault,
+) *PublicParamsManager[T] {
+	return &PublicParamsManager[T]{
+		Logger:                   logger,
+		Vault:                    vault,
+		PPLabel:                  PPLabel,
+		PublicParamsDeserializer: PublicParamsDeserializer,
+	}
+}
+
+func NewPublicParamsManagerFromParams[T driver.PublicParameters](logger *flogging.FabricLogger, PublicParamsDeserializer PublicParamsDeserializer[T], pp T) (*PublicParamsManager[T], error) {
+	return &PublicParamsManager[T]{
+		Logger:                   logger,
+		publicParameters:         pp,
+		PublicParamsDeserializer: PublicParamsDeserializer,
+	}, nil
+}
+
+func (v *PublicParamsManager[T]) PublicParameters() driver.PublicParameters {
+	v.Mutex.RLock()
+	defer v.Mutex.RUnlock()
+
+	return v.publicParameters
+}
+
+// SerializePublicParameters returns the public params in a serialized form
+func (v *PublicParamsManager[T]) SerializePublicParameters() ([]byte, error) {
+	return v.PublicParams().Serialize()
+}
+
+func (v *PublicParamsManager[T]) NewCertifierKeyPair() ([]byte, []byte, error) {
+	return nil, nil, errors.Errorf("not supported")
+}
+
+func (v *PublicParamsManager[T]) Load() error {
+	ppRaw, err := v.Vault.PublicParams()
+	if err != nil {
+		return errors.WithMessage(err, "failed to fetch public params from vault")
+	}
+	if len(ppRaw) == 0 {
+		return nil
+	}
+
+	v.Logger.Debugf("fetched public parameters [%s], set them...", hash.Hashable(ppRaw).String())
+	return v.SetPublicParameters(ppRaw)
+}
+
+// SetPublicParameters updates the public parameters with the passed value
+func (v *PublicParamsManager[T]) SetPublicParameters(raw []byte) error {
+	v.Mutex.Lock()
+	defer v.Mutex.Unlock()
+
+	if len(raw) == 0 {
+		return errors.Errorf("empty public parameters")
+	}
+
+	pp, err := v.PublicParamsDeserializer.DeserializePublicParams(raw, v.PPLabel)
+	if err != nil {
+		return err
+	}
+
+	if err := pp.Validate(); err != nil {
+		return errors.WithMessage(err, "invalid public parameters")
+	}
+
+	for _, callback := range v.Callbacks {
+		if err := callback(pp); err != nil {
+			return err
+		}
+	}
+
+	v.Logger.Debugf("set public parameters [%s]", hash.Hashable(raw).String())
+	v.publicParameters = pp
+
+	return nil
+}
+
+func (v *PublicParamsManager[T]) PublicParams() T {
```

Comment:
This method is a duplicate I think

Reply:
there is an interface that needs to be satisfied and requires a function that returns driver.PublicParamters